server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -1389,7 +1398,7 @@ public void finalizeSnapshot(final FinalizeSnapshotContext finalizeSnapshotConte
             new SnapshotException(metadata.name(), snapshotId, "failed to update snapshot in repository", e)
         );
 
-        final Executor executor = threadPool.executor(ThreadPool.Names.SNAPSHOT);
+        final Executor executor = threadPool.executor(ThreadPool.Names.SNAPSHOT_META);
```

Comment:
:+1: definitely makes sense to me to parallelise harder the writing of updated metadata when creating a snapshot

Reply:
The risk here though, is that now a massive parallelisation (think first snapshot of 5k indices, writing 5k metadata files) will completely block the get-snapshots API or snapshot status APIs until it's through. I feel like that's an issue since we and Cloud won't be able to get a snapshot's status for an extended period of time potentially.

Reply:
Ah ok that's a good point. Should we throttle here to keep the real queue size under control, or is this whole idea just not going to work?

Reply:
I think if we want this, yes then we need to have some mechanism like the `ThrottledTaskRunner` in place to prevent from starvation on this pool.
I really do wonder if it's worth it messing with this though? In the end, if we really want finalisation faster we should introduce batching to it. It's really rare for finalisation to be slow on uploading index metadata ever since we deduplicate per-index for unchanged metadata, so this really only hits the first snapshot in a large cluster. 
Finalisation tends to just be slow since it's this multi-step process including CS updates and writing the heavy `RepositoryData` in the real world I think.

Reply:
I thought something like this would have helped in the SDHE linked from https://github.com/elastic/elasticsearch/issues/93220 but I'm not convinced about that any more.

Still, this seems like a much smaller change than batched finalisations.

Reply:
++ I'm not so sure this will be of much help with that issue. Intuitively I'd rather not break with the separate pools for read and write and have to mess with not blocking the pool in tricky ways unless we get a clear win from it.

Reply:
ok, sorry @pxsalehi for the wild-goose chase here :)

Reply:
No problem! :-)